vlm/src/main/scala/geotrellis/contrib/vlm/gdal/URIFormatter.scala

```diff
@@ -0,0 +1,91 @@
+package geotrellis.contrib.vlm.gdal
+
+
+import java.net.URI
+
+
+object Schemes {
+  final val FTP = "ftp"
+  final val HTTP = "http"
+  final val HTTPS = "https"
+
+  final val TAR = "tar"
+  final val ZIP = "zip"
+  final val GZIP = "gzip"
+
+  final val FILE = "file"
+
+  final val S3 = "s3"
+
+  final val GS = "gs"
+
+  final val WASB = "wasb"
+  final val WASBS = "wasbs"
+
+  final val HDFS = "hdfs"
+
+  final val FILE_TYPE_TO_SCHEME =
+    Map(
+      TAR -> TAR,
+      "tgz" -> TAR,
+      ZIP -> ZIP,
+      "kmz" -> ZIP,
+      "ods" -> ZIP,
+      "xlsx" -> ZIP,
+      GZIP -> GZIP
+    )
+}
+
+
+object URIFormatter {
+  import Schemes._
+
+  def apply(path: String): String =
+    apply(new URI(path))
+
+  def apply(uri: URI): String = {
+    // TODO: Support reading a single file from a zip file
+    // ie. reading 1.tiff from a zip: s3://bucket/path/to/my/zipped/data/data.zip/1.tiff
+
+    val path = uri.toString
+
+    // We must first determine if we need to chain file handlers
+    // ie. Reading from a zip file on s3
+    val leadingScheme: String =
+      if (path.contains("."))
+        FILE_TYPE_TO_SCHEME.get(path.substring(path.lastIndexOf(".") + 1)) match {
```

Comment:
can it handle smth [like that](https://github.com/mapbox/rasterio/blob/960a906dad2a4e426387ce048a52c6e90afdcd2b/rasterio/path.py#L45) ?